### DISCUSSION TITLE
Shelly 1PM overheating, Tasmota 12.5.0

### DISCUSSION THREAD

**Discussion by bastian987** [arendst/Tasmota#19475] [General]
Hi,
I use Tasmota on many devices and it runs great! However, I have regular issues with one setup:
I want to use two Shelly 1pm to track the energy consumption of my washer and dryer, e.g., to send me notifications when the cycle is completed. I run the Shellys on Tasmota 12.5.0, and when running the washer, the temperature of the Shelly gets high, from time to time to the level that Tasmota shuts down the device. Room temperature is rather cool in my cellar, I even tried attaching small heat deflectors to the Shelly — without much success. Operation is definitely in spec of the Shelly, there are no other devices attached than the washer to one Shelly and the dryer to another. Repeating the installation (to make sure the contact of the screws is fine) didn’t help.

What am I doing wrong? Shelly 1pm not suitable for a device with maybe 2kW peak? Issue with Tasmota? Incorrect installation, too cramped?
![IMG_7353](https://github.com/arendst/Tasmota/assets/44191521/fbec09d5-a794-4229-ae16-8221f26711f6)

**Comment by stefanbode** [comment]
I have quite a similar problem with the shelly plus 2PM. This esp32 based also run quite high on temperature and I'm not sure if just the measurement is wrong. I use the device to measure solar power. this is 0..600W. (Very low!!!) and still at 600W the temperature rises up to 120° in the UI. I measure with a thermo stick and do not see more than 60° at this point. I do also not see any melting. As soon as I switch it OFF the temperature goes down. At the end it works, but is a bit scary...

**Reply by stefanbode** [reply]
![image](https://github.com/arendst/Tasmota/assets/16225290/ce67dfab-5e9a-41c7-9cc4-65985299b4bb)

**Reply by bastian987** [reply]
Thanks for sharing. Would be interesting to know if the temperature measurement is incorrect and that’s the actual root cause. I could measure the Shelly surface temperature, but I don’t think this value would be of much help. As a reference: another 1pm I‘m running (still on 6.6.0) shows 40 degrees Celsius when idling, not sure this value looks reasonable.

BTW  @stefanbode I‘m still running your branch to control our all roller shutters, big thanks for the contribution at the time! :)

**Reply by sfromis** [reply]
The Shelly 2.5 PM (as shown by stefanbode) is known to run rather hot on the inside, especially when on. According to Shelly themselves, around 90 °C is normal. However, the Shelly 1PM is a different device.

And also the Shelly Plus 2PM text-mentioned by stefanbode is a very different device again, as I don't think that anyone has made sure to have the temperature sensor for that one calibrated. The ESP32 temperature of that one (if not disabled) is known to be **way** off anything real.

**Reply by stefanbode** [reply]
As said up to 120°C is possible. right now also in the way I see it very often:
![image](https://github.com/arendst/Tasmota/assets/16225290/67bb16d5-a6a6-432f-a47a-cdee41cfd533)

@bastian987 : regarding shutters you should switch back to the main branch. since a year or two all development is directly done in the main project. quite rare experiments in my branch.

**Comment by bastian987** [comment]
But the 90 degree Celsius followed by automatically turning off the output is not expected behavior for the 1pm, right? Devices like a washing machine should be possible behind a 1pm Shelly with Tasmota? For me, it’s unclear — is it hardware or software related? What’s a potential path to fix the issue? Any hints are highly appreciated.

**Reply by sfromis** [reply]
Not clear what you mean by "the issue".....

`SetOption42` controls over-temperature threshold, default 90°C.

**Reply by bastian987** [reply]
Ok, let me try to rephrase:
I was working under the assumption, the temperature of the Shelly in normal operation should not go above the default Tasmota threshold. I have read in the Shelly forums that the original firmware shuts the device down at 95 degree Celsius.

I would have expected the default value in Tasmota, which is similar to the one in the original firmware (+5), ensures there is no fire hazard. Is the recommended, safe solution to increase the temperature threshold? If this is not safe, the conclusion would be a 1pm with Tasmota cannot be used with a device like a washing machine.

**Reply by sfromis** [reply]
My first question would be if the Tasmota config for the internal Shelly 1PM temperature sensor is actually correct. While I have no reason to assume that it is not, I'd not automatically expect the value to be correct. An analog temp sensor has a set of values which should match the specific electrical components, but if not...... 

I'd expect the default for the [ADC](https://tasmota.github.io/docs/ADC/) temperature pin to be to be `AdcParam 2,32000,10000,3350`, and I seem to remember that the defaults were chosen to match some Shelly device. That was a long time ago, and sometimes hardware revisions need a different config.

Comments in the analog driver suggest that the defaults were set to match the Shelly 2.5, which may or may not be similar to the Shelly 1PM, and what was chosen years ago may or may not be valid now.

Yeah, I know, this just makes it even more confusing, but simple answers may be too much guesswork.

**Reply by joba-1** [reply]
the default value in Tasmota ensures no fire hazard and does not switch off unnecessarily only under the assumption the reported temperature is correct (and some more assumptions, but let’s not get too picky).

I know MY shelly sensor does NOT fulfill this requirement, it reports way too high values.

If you want to know if YOUR shelly fulfills it, let it run idle in an oven set to 90°C for 20 min. It should report an only slightly higher value. 
If it doesn’t, using the reported value with setoption42 should be ok (with the assumption that sensor values for the same temperatures are always more or less the same: so better repeat the test)

I give no guarantee the above will really work :)

**Reply by sfromis** [reply]
Another test could to to have it unpowered for some time, and check the reported temperature immediately after connecting power again. Note that, depending on components and config values, reasonable readings at one temperature level does not mean that it can be assumed to also be reasonable for a very different temperature level. Trying to test it should probably both be at "low" and "high" temperature.
